mysite/announcements/tasks.py

```diff
@@ -40,3 +44,5 @@ def expire_announcements():
         expired_announcement_id.append(announcement.id)
 
     Announcements.objects.filter(id__in=expired_announcement_id).update(has_expired=True)
+    Status.objects.filter(announcement_id__in=expired_announcement_id, status='yet to receive')\
+        .update(status='expired', status_last_update_time=current_time)
```

Comment:
Since you are using string values here, now there are chances of errors due to situations such as:- 
- What if I change a string in models.py? I would need to remember to change here too. And as code gets big this problem also gets bigger.


So, it is best to define things as such(which you plan to use across the Django app) in a file such as constants.py and import and use it across.